__tests__/app.test.js

```diff
@@ -270,4 +269,76 @@ describe("GET /api/articles/:article_id/comments", () => {
         });
     });
   });
+
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("POST 200: posts the user's comment", () => {
+      const newComment = { username: "butter_bridge", comment: "test comment" };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({ body }) => {
+          const { postedComment } = body;
+          expect(postedComment).toBe(newComment.comment);
+        });
+    });
+
+    test("POST 400: returns an error message stating that the user is missing input data when no username", () => {
+      const newCommentNoUser = { comment: "test comment" };
+
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newCommentNoUser)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Something wrong with body");
+        });
+    });
+
+    test("POST 400: returns an error message stating that the yser is missing input data when no comment", () => {
+      const newCommentNoComment = { username: "testUser" };
+
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newCommentNoComment)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Something wrong with body");
+        });
+    });
+
+    test("POST 400: returns an error message when trying to post a comment at an invalid id", () => {
+      const newComment = { username: "testUser", comment: "test comment" };
+      return request(app)
+        .post("/api/articles/1000/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("no article found at id given");
+        });
+    });
+
+    test("POST 404: returns an error message when user does not exist", () => {
+      const newComment = { username: "testUser", comment: "test comment" };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("username not found");
+        });
+    });
+
+    test("POST 404: returns an error message when a user tries to add a comment at an id that does not exist", () => {
```

Comment:
400

Reply:
this description doesnt match your test, you do need to test for this also ie "api/articles/999/comments"

Reply:
You are testing for the 400 but you haven't changed the description here too